src/srcsets.ts

```diff
@@ -0,0 +1,90 @@
+// ----- Imports ----- //
+
+import { createHash } from "crypto";
+import type { Result } from "@guardian/types";
+import { fromUnsafe, ResultKind } from "@guardian/types";
+
+// ----- Setup ----- //
+
+const imageResizer = "https://i.guim.co.uk/img";
+
+const defaultWidths = [140, 500, 1000, 1500, 2000];
```

Comment:
How are these widths chosen?

Reply:
They suspiciously look like default Grid widths created for every crop as long as it’s big enough. These statis image crops should never ever be used with the resizer. Only master crop should be used. VCL has a clever logic for when resizer is down to mangle URLs and display these crops.

Unless I’m mistaken, I haven’t looked beyond this line of code!

Reply:
Indeed, the widths we get on DCR from Frontend often (not always) correspond to the exact size we're asking for in sizes which seems like a worthy goal.

These are all the exact widths we ask for:

140
300
620
660
700
780
860
940
1020
1300
(we don't ask for it but we would want a larger image here, 1900 I think)

We also use `vw` or `vh` to request a width based on viewport in some situations.

But a lot of these are context specific, only used for immersives, or showcase articles for example. Could we let the consumer ask for the list of sources dynamically based on state?

Perhaps this is already the case but maybe we should have the consumer pass in the widths they want?

Reply:
So the widths Oliver lists come from [ImageProfile.scala](https://github.com/guardian/frontend/blob/main/common/app/views/support/ImageProfile.scala#L110-L124), I think. Conceptually, they should reflect exact designed image sizes on all breakpoints in CSS pixels. While it’s possible that they have (?) a secondary reason, steering designers towards more consistency in their work, in theory at least, they wouldn’t be hardcoded but would come straight from CSS pixel widths (and, one day, heights) in the designs themselves (so eg may differ by product). And when the designs change, they should change too.

This doesn’t apply to

> We also use vw or vh to request a width based on viewport in some situations.

which relates to “fluid” (or “immersive”) imagery whose sizes come from devices per breakpoint. I also suspect

>  (we don't ask for it but we would want a larger image here, 1900 I think)

is also for a “fluid” image use (?).

These “fluid” images may need slightly different responsive steps strategy. I think.

Reply:
> Only master crop should be used.

Don't worry, it is. I just re-used the same values here because I needed a reasonable list of sizes, but they're not set in stone. We can change it to the list @oliverlloyd provided above, or any list you choose @paperboyo.